src/logic/algo/kernels.ts

```diff
@@ -2,4 +2,8 @@ export const ubcKernel = (
   meanNodeValue: number,
   visitsOfNode: number,
   visitsOfParent: number
-): number => 0;
+): number => {
+  const explorationWeight = 0.1;
+  const logVisitsOfParent = Math.log(visitsOfParent);
+  return meanNodeValue + explorationWeight * Math.sqrt(logVisitsOfParent / visitsOfNode);
```

Comment:
Może być tak, że `visitsOfNode` wyniesie 0?

Reply:
Teoretycznie nie, bo jak dodajemy node do słownika nodeVisits to przypisujemy mu wartość jeden, a jeśli by go tam nie było (co  nie powinno się zdarzyć) to domyślnie w do funkcji ubcKernel brane jest 1,  ale dodam tu obsługę błędu na wszelki wypadek.